code/drasil-gool/GOOL/Drasil/LanguageRenderer/CppRenderer.hs

```diff
@@ -286,8 +284,8 @@ instance (Pair p) => InternalVarElim (p CppSrcCode CppHdrCode) where
   variable v = RC.variable $ pfst v
 
 instance (Pair p) => RenderVariable (p CppSrcCode CppHdrCode) where
-  varFromData b n t d = pair (varFromData b n (pfst t) d) 
-    (varFromData b n (psnd t) d)
+  varFromData b n t' d = pair1 (\t ->varFromData b n t d) 
```

Comment:
This pattern (of having to introduce a lambda) often [but not always] means that the order of arguments in `varFromData` is sub-optimal, and it would be better if `t` was last. This should be investigated.